src/js/actions.ts

```diff
@@ -104,7 +104,7 @@ export default class Actions {
           day.manipulate(1, Unit.month);
         }
 
-        day.date = +currentTarget.innerText;
+        day.date = +currentTarget.getAttribute('data-day');
```

Comment:
In case you didn't know this, to quote [MDN](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes#javascript_access): 
> Reading the values of these attributes out in JavaScript is also very simple. You could use getAttribute() with their full HTML name to read them, but the standard defines a simpler way: a DOMStringMap you can read out via a dataset property.

in this case, that would be:
```suggestion
        day.date = +currentTarget.dataset.day;
```